### PR TITLE
Grid navigation enhancement

### PR DESCRIPTION
**Pull request by prakash1998**

### CODE REVIEW

File: src/selection-util.js [hunk 1]
```diff
@@ -23,24 +23,42 @@ export const isRowSelected = (rowIndex, selection) => {
...
   const { x1, y1, x2, y2 } = selection
 
   if (!expand) {
-    const x = Math.max(x1 - 1, 0)
-    return { x1: x, x2: x, y1, y2: y1 }
+    let x = x1 - 1
+    let y = y1
+    if (x < 0) {
+      x = colCount - 1
+      if (y > 0) {
```
Comment (fobbyal) [inline src/selection-util.js:34]: i think there might be a bug here where the `y < 0` case is only taken care of when `x < 0`.

File: src/selection-util.js [hunk 2]
```diff
@@ -23,24 +23,42 @@ export const isRowSelected = (rowIndex, selection) => {
...
   const { x1, y1, x2, y2 } = selection
 
   if (!expand) {
-    const x = Math.min(x1 + 1, colCount - 1)
-    return { x1: x, x2: x, y1, y2: y1 }
+    let x = x1 + 1
+    let y = y1
+    if (x > colCount - 1) {
+      x = 0
+      if (y < rowCount - 1) {
```
Comment (fobbyal) [inline src/selection-util.js:55]: same as comment before. I believe the y and x values should be kept in bound independently

File: src/selection-util.js (hunk 1 above)
Comment (prakash1998) [reply src/selection-util.js:34]: on left underflow, need to move selection in only up direction, so when underflow occur, just need to check for y>0

File: src/selection-util.js (hunk 2 above)
Comment (prakash1998) [reply src/selection-util.js:55]: on right overflow, need to move selection in only down direction, so when overflow occur, just need to check for y < maxRow

**Pull request merged**